apps/api/views/metrics.py

```diff
@@ -0,0 +1,761 @@
+        }
+
+        instructors_data = {
+            'data': list(instructors),
+            'unique_pairs': {
+                'instructors_join_count': 0,
+            },
+            'sorted': True,
+        }
+
+        students_data = {
+            'data': list(students),
+            'unique_pairs': {
+                'students_join_count': 0,
+            },
+            'sorted': True,
+        }
+
+        # Find union of instructors and students based on date
+        merged_time_series_data = union_lists('date', instructors_data, students_data)
+
+        if len(merged_time_series_data) > 0:
+            merged_time_series_data[0].update(stats)
+        else:
+            merged_time_series_data.append(stats)
+
+        serializer = AdminMetricsSerializer(data=merged_time_series_data, many=True)
+        serializer.is_valid(raise_exception=True)
+        return Response(serializer.validated_data)
+
+
+class AdminKlassCSVView(TimeSeriesObjectCreationQueryMixin, APIView):
+    permission_classes = (InstructorOrSuperuserPermission,)
+    renderer_classes = (AdminKlassCSVRenderer,)
+
+    time_series_model = Klass
+    time_series_creation_date_field_name = 'created'
+
+    def get(self, request, **kwargs):
+        klasses = self.time_series_query(count_field_name='klasses_created_count')
+
+        class Round(Func):
+            function = 'ROUND'
+            arity = 2
+
+        klasses_total = Klass.objects.count()
+        ave_students_per_klass = Klass.objects.all().annotate(student_count=Count('enrolled_students')).aggregate(ave_students=Round(Avg('student_count'), 2))
+        ave_subs_per_definition = Definition.objects.all().annotate(submission_count=Count('submissions')).aggregate(ave_subs=Round(Avg('submission_count'), 2))
+        ave_definitions_per_klass = Klass.objects.all().annotate(definition_count=Count('homework_definitions')).aggregate(ave_definitions=Round(Avg('definition_count'), 2))
+        data = list(klasses)
+        stats = {
+            'klasses_total': klasses_total,
+            'ave_students_per_klass': ave_students_per_klass.get('ave_students'),
+            'ave_definitions_per_klass': ave_definitions_per_klass.get('ave_definitions'),
+            'ave_submissions_per_definition': ave_subs_per_definition.get('ave_subs'),
+        }
+        if len(data) > 0:
+            data[0].update(stats)
+        else:
+            data.append(stats)
+
+        serializer = AdminMetricsSerializer(data=data, many=True)
+        serializer.is_valid(raise_exception=True)
+        return Response(serializer.validated_data)
+
+
+class AdminSubmissionCSVView(TimeSeriesObjectCreationQueryMixin, ScoreDistributionMixin, APIView):
+    permission_classes = (InstructorOrSuperuserPermission,)
+    renderer_classes = (AdminSubmissionCSVRenderer,)
+
+    time_series_model = Submission
+    time_series_creation_date_field_name = 'created'
+
+    def get(self, request, **kwargs):
```

Comment:
Just checking that `APIView` doesn't have this defined by default? It's just a base API View right versus viewsets that have everything already?

Reply:
I think I know what you're saying. `APIView` requires you to explicitly write the methods that you want to work on the view, as opposed to a viewset that has all of the methods by default, and is generally configured via queryset. Does this answer your question, or am I missing the point completely?